## 1. The oldest businesses in the world
<p>This is Staffelter Hof Winery, Germany's oldest business, which was established in 862 under the Carolingian dynasty. It has continued to serve customers through dramatic changes in Europe such as the Holy Roman Empire, the Ottoman Empire, and both world wars. What characteristics enable a business to stand the test of time? Image credit: <a href="https://commons.wikimedia.org/wiki/File:MKn_Staffelter_Hof.jpg">Martin Kraft</a>
<img src="https://assets.datacamp.com/production/project_1383/./img/MKn_Staffelter_Hof.jpeg" alt="The entrance to Staffelter Hof Winery, a German winery established in 862." width="400"></p>
<p>To help answer this question, BusinessFinancing.co.uk <a href="https://businessfinancing.co.uk/the-oldest-company-in-almost-every-country">researched</a> the oldest company that is still in business in almost every country and compiled the results into a dataset. Let's explore this work to to better understand these historic businesses. Our datasets, which are all located in the <code>datasets</code> directory, contain the following information: </p>
<h3 id="businessesandnew_businesses"><code>businesses</code> and <code>new_businesses</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>varchar</td>
<td>Name of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Year the business was founded.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>char</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
</tbody>
</table>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>varchar</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>varchar</td>
<td>Name of the country.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>varchar</td>
<td>Name of the continent that the country exists in.</td>
</tr>
</tbody>
</table>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>varchar</td>
<td>Description of the business category.</td>
</tr>
</tbody>
</table>
<p>Now let's learn about some of the world's oldest businesses still in operation!</p>

In [168]:
# Import the pandas library under its usual alias
import pandas as pd

# Load the business.csv file as a DataFrame called businesses
businesses = pd.read_csv("datasets/businesses.csv")

# Sort businesses from oldest businesses to youngest
sorted_businesses = businesses.sort_values("year_founded")

# Display the first few lines of sorted_businesses

print(sorted_businesses.shape[0])
sorted_businesses.head()


163


business  year_founded category_code country_code
64                    Kongō Gumi           578          CAT6          JPN
94   St. Peter Stifts Kulinarium           803          CAT4          AUT
107        Staffelter Hof Winery           862          CAT9          DEU
106            Monnaie de Paris            864         CAT12          FRA
103               The Royal Mint           886         CAT12          GBR

## 2. The oldest businesses in North America
<p>So far we've learned that Kongō Gumi is the world's oldest continuously operating business, beating out the second oldest business by well over 100 years! It's a little hard to read the country codes, though. Wouldn't it be nice if we had a list of country names to go along with the country codes?</p>
<p>Enter <code>countries.csv</code>, which is also located in the <code>datasets</code> folder. Having useful information in different files is a common problem: for data storage, it's better to keep different types of data separate, but for analysis, we want all the data in one place. To solve this, we'll have to join the two tables together. </p>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>varchar</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>varchar</td>
<td>Name of the country.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>varchar</td>
<td>Name of the continent that the country exists in.</td>
</tr>
</tbody>
</table>
<p>Since <code>countries.csv</code> contains a <code>continent</code> column, merging the datasets will also allow us to look at the oldest business on each continent! </p>

In [170]:
# Load countries.csv to a DataFrame
countries = pd.read_csv("datasets/countries.csv")

# Merge sorted_businesses with countries
businesses_countries = sorted_businesses.merge(countries , on='country_code')
businesses_countries


business  year_founded  \
0                                        Kongō Gumi           578   
1                       St. Peter Stifts Kulinarium           803   
2                             Staffelter Hof Winery           862   
3                                 Monnaie de Paris            864   
4                                    The Royal Mint           886   
5                                        Sean's Bar           900   
6                            Marinelli Bell Foundry          1040   
7                                  Affligem Brewery          1074   
8                                       Munke Mølle          1135   
9                Ma Yu Ching's Bucket Chicken House          1153   
10                                Casa de Ganaderos          1218   
11                                  Gasthof Sternen          1230   
12                                Bochnia Salt Mine          1248   
13                             Drohobych salt plant          1250   
14                                    Kremnica Mint          1328   
15                                The Brand Brewery          1340   
16                                  Skyllbergs bruk          1346   
17                                  Pivovar Broumov          1348   
18                                        Raeapteek          1422   
19                                  Gostilna Gastuž          1467   
20                                           Mousel          1511   
21                      CTT-Correios de Portugal SA          1520   
22                      La Casa de Moneda de México          1534   
23                          Casa Nacional de Moneda          1565   
24                               Çemberlitaş Hamamı          1584   
25                                        Cēsu Alus          1590   
26                                     Piraeus Bank          1606   
27                       Casa de Moneda de Colombia          1621   
28                               Shirley Plantation          1638   
29                                     Posten Norge          1647   
..                                              ...           ...   
133                Cameroon Development Corporation          1947   
134                       Myanmar National Airlines          1948   
135          Salam International Investment Limited          1952   
136                       National Bank of Cambodia          1954   
137                                         Brarudi          1955   
138     Office de Radiodiffusion-Télévision du Mali          1957   
139                                     Tashi Group          1959   
140                             Electricite du Laos          1959   
141         Office of Radio and Television of Niger          1960   
142                                    Halco Mining          1962   
143                                 Yemenia Airways          1962   
144                                  Air Madagascar          1962   
145                                      Swazi Rail          1963   
146                 Malawi Broadcasting Corporation          1964   
147                        Botswana Meat Commission          1965   
148                                     Air Burkina          1967   
149                  Communauté Électrique du Bénin          1968   
150                                  BGFIBank Group          1971   
151                                      Cotontchad          1971   
152                      Central Bank of Mauritania          1973   
153                        Correios da Guiné-Bissau          1973   
154           Central Bank of São Tomé and Príncipe          1975   
155  Development Bank of the Central African States          1975   
156                                  Air Seychelles          1977   
157                         Central Bank of Lesotho          1978   
158                     Central Bank of the Comoros          1981   
159                          European Tru

In [171]:
# Import the pandas library under its usual alias
import pandas as pd

# Load the business.csv file as a DataFrame called businesses
businesses = pd.read_csv("datasets/businesses.csv")
# Load countries.csv to a DataFrame
countries = pd.read_csv("datasets/countries.csv")

# Merge sorted_businesses with countries
businesses_countries = sorted_businesses.merge(countries , on='country_code')

# Filter businesses_countries to include countries in North America only
north_america = businesses_countries[businesses_countries['continent']== 'North America']
print(north_america.shape[0])
north_america.head()


17


business  year_founded category_code country_code  \
22  La Casa de Moneda de México          1534         CAT12          MEX   
28           Shirley Plantation          1638          CAT1          USA   
33         Hudson's Bay Company          1670         CAT17          CAN   
35                Mount Gay Rum          1703          CAT9          BRB   
40                    Rose Hall          1770         CAT19          JAM   

          country      continent  
22         Mexico  North America  
28  United States  North America  
33         Canada  North America  
35       Barbados  North America  
40        Jamaica  North America

## 3. The oldest business on each continent
<p>Now we can see that the oldest company in North America is La Casa de Moneda de México, founded in 1534. Why stop there, though, when we could easily find out the oldest business on every continent? </p>

In [173]:
# Create continent, which lists only the continent and oldest year_founded
continent = businesses_countries.groupby("continent").agg({"year_founded":"min"})

# Merge continent with businesses_countries
merged_continent = continent.merge(businesses_countries, on=["continent", "year_founded"])

# Subset continent so that only the four columns of interest are included
subset_merged_continent = merged_continent[["continent", "country", "business", "year_founded"]]
subset_merged_continent

continent    country                     business  year_founded
0         Africa  Mauritius               Mauritius Post          1772
1           Asia      Japan                   Kongō Gumi           578
2         Europe    Austria  St. Peter Stifts Kulinarium           803
3  North America     Mexico  La Casa de Moneda de México          1534
4        Oceania  Australia               Australia Post          1809
5  South America       Peru      Casa Nacional de Moneda          1565

In [174]:
# Create continent, which lists only the continent and oldest year_founded
continent = businesses_countries.groupby('continent').agg({'year_founded':'min'})

continent

year_founded
continent                  
Africa                 1772
Asia                    578
Europe                  803
North America          1534
Oceania                1809
South America          1565

In [176]:

# Merge continent with businesses_countries
merged_continent = continent.merge(businesses_countries,on=["continent", "year_founded"])
merged_continent


continent  year_founded                     business category_code  \
0         Africa          1772               Mauritius Post         CAT16   
1           Asia           578                   Kongō Gumi          CAT6   
2         Europe           803  St. Peter Stifts Kulinarium          CAT4   
3  North America          1534  La Casa de Moneda de México         CAT12   
4        Oceania          1809               Australia Post         CAT16   
5  South America          1565      Casa Nacional de Moneda          CAT3   

  country_code    country  
0          MUS  Mauritius  
1          JPN      Japan  
2          AUT    Austria  
3          MEX     Mexico  
4          AUS  Australia  
5          PER       Peru

In [177]:
# Subset continent so that only the four columns of interest are included
subset_merged_continent =merged_continent[['continent', 'country', 'business', 'year_founded']]
subset_merged_continent

continent    country                     business  year_founded
0         Africa  Mauritius               Mauritius Post          1772
1           Asia      Japan                   Kongō Gumi           578
2         Europe    Austria  St. Peter Stifts Kulinarium           803
3  North America     Mexico  La Casa de Moneda de México          1534
4        Oceania  Australia               Australia Post          1809
5  South America       Peru      Casa Nacional de Moneda          1565

## 4. Unknown oldest businesses
<p>BusinessFinancing.co.uk wasn't able to determine the oldest business for some countries, and those countries are simply left off of <code>businesses.csv</code> and, by extension, <code>businesses</code>. However, the <code>countries</code> that we created <em>does</em> include all countries in the world, regardless of whether the oldest business is known. </p>
<p>We can compare the two datasets in one DataFrame to find out which countries don't have a known oldest business! </p>

In [178]:

all_countries = businesses.merge(countries,on='country_code', how='right', indicator=True)
# Filter to include only countries without oldest businesses
missing_countries = all_countries[all_countries['_merge'] != 'both' ]

# Create a series of the country names with missing oldest business data
missing_countries_series = missing_countries['country']

# Display the series
missing_countries_series
print(all_countries)

                                           business  year_founded  \
0                                    Hamoud Boualem        1878.0   
1                    Communauté Électrique du Bénin        1968.0   
2                          Botswana Meat Commission        1965.0   
3                                       Air Burkina        1967.0   
4                                           Brarudi        1955.0   
5                  Cameroon Development Corporation        1947.0   
6                            Correios de Cabo Verde        1849.0   
7        Banque Internationale pour la Centrafrique        1946.0   
8                                        Cotontchad        1971.0   
9                       Central Bank of the Comoros        1981.0   
10    Société nationale des Chemins de fer du Congo        1889.0   
11   Development Bank of the Central African States        1975.0   
12                                         La Poste        1945.0   
13                          Ethio-

In [179]:
import pandas as pd
# Load the business.csv file as a DataFrame called businesses
businesses = pd.read_csv("datasets/businesses.csv")
# Load countries.csv to a DataFrame
countries = pd.read_csv("datasets/countries.csv")
# Use .merge() to create a DataFrame, all_countries
all_countries = businesses.merge(countries, on="country_code", how="right",  indicator=True)
# Filter to include only countries without oldest businesses
missing_countries = all_countries[all_countries["_merge"] != "both"]

# Create a series of the country names with missing oldest business data
missing_countries_series = missing_countries["country"]

# Display the series
missing_countries_series


163                              Angola
164                 Antigua and Barbuda
165                             Bahamas
166                  Dominican Republic
167                             Ecuador
168                                Fiji
169     Micronesia, Federated States of
170                               Ghana
171                              Gambia
172                             Grenada
173           Iran, Islamic Republic of
174                          Kyrgyzstan
175                            Kiribati
176               Saint Kitts and Nevis
177                              Monaco
178                Moldova, Republic of
179                            Maldives
180                    Marshall Islands
181                               Nauru
182                               Palau
183                    Papua New Guinea
184                            Paraguay
185                 Palestine, State of
186                     Solomon Islands
187                            Suriname


## 5. Adding new oldest business data
<p>It looks like we've got some holes in our dataset! Fortunately, we've taken it upon ourselves to improve upon BusinessFinancing.co.uk's work and find oldest businesses in a few of the missing countries. We've stored the newfound oldest businesses in <code>new_businesses</code>, located at <code>"datasets/new_businesses.csv"</code>. It has the exact same structure as our <code>businesses</code> dataset. </p>
<h3 id="new_businesses"><code>new_businesses</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>varchar</td>
<td>Name of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Year the business was founded.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>char</td>
<td>ISO 3166-1 3-letter country code.</td>
</tr>
</tbody>
</table>
<p>All we have to do is combine the two so that we've got one more complete list of businesses!</p>

In [181]:
# Import new_businesses.csv
new_businesses = pd.read_csv("datasets/new_businesses.csv")

# Add the data in new_businesses to the existing businesses
all_businesses = pd.concat([new_businesses, businesses])

# Merge and filter to find countries with missing business data
new_all_countries = all_businesses.merge(countries, on="country_code", how="outer",  indicator=True)
new_missing_countries = new_all_countries[new_all_countries["_merge"] != "both"]

# Group by continent and create a "count_missing" column
count_missing = new_missing_countries.groupby("continent").agg({"country":"count"})
count_missing.columns = ["count_missing"]
count_missing

count_missing
continent                   
Africa                     3
Asia                       7
Europe                     2
North America              5
Oceania                   10
South America              3

In [182]:
# Import new_businesses.csv
new_businesses = pd.read_csv("datasets/new_businesses.csv")
new_businesses


business  year_founded category_code country_code
0                 Fiji Times          1869         CAT13          FJI
1  J. Armando Bermúdez & Co.          1852          CAT9          DOM

In [183]:
# Add the data in new_businesses to the existing businesses
all_businesses = pd.concat([new_businesses, businesses])
all_businesses

business  year_founded  \
0                                        Fiji Times          1869   
1                         J. Armando Bermúdez & Co.          1852   
0                                    Hamoud Boualem          1878   
1                    Communauté Électrique du Bénin          1968   
2                          Botswana Meat Commission          1965   
3                                       Air Burkina          1967   
4                                           Brarudi          1955   
5                  Cameroon Development Corporation          1947   
6                            Correios de Cabo Verde          1849   
7        Banque Internationale pour la Centrafrique          1946   
8                                        Cotontchad          1971   
9                       Central Bank of the Comoros          1981   
10    Société nationale des Chemins de fer du Congo          1889   
11   Development Bank of the Central African States          1975   
12                                         La Poste          1945   
13                          Ethio-Djibouti Railways          1901   
14                       Egyptian National Railways          1854   
15                       Guinea Ecuatorial Airlines          1996   
16                                   Asmara Brewery          1939   
17                        National Bank of Ethiopia          1906   
18                                   BGFIBank Group          1971   
19                                     Halco Mining          1962   
20                         Correios da Guiné-Bissau          1973   
21                                KCB Group Limited          1896   
22                          Central Bank of Lesotho          1978   
23                          National Port Authority          1921   
24                                        Umma Bank          1907   
25                                   Air Madagascar          1962   
26                  Malawi Broadcasting Corporation          1964   
27      Office de Radiodiffusion-Télévision du Mali          1957   
..                                              ...           ...   
133                            Drohobych salt plant          1250   
134                                   Mount Gay Rum          1703   
135                                     Belize Bank          1902   
136                            Hudson's Bay Company          1670   
137                    Florida Ice and Farm Company          1908   
138                              Cubana de Aviación          1929   
139                        The Chronicle (Dominica)          1909   
140                                HSBC El Salvador          1891   
141                   Corporación Multi Inversiones          1920   
142                                Rhum Barbancourt          1862   
143                   National Railroad of Honduras          1870   
144                                       Rose Hall          1770   
145                     La Casa de Moneda de México          1534   
146                                    Flor de Caña          1890   
147                         National Bank of Panama          1904   
148                   1st National Bank of St Lucia          1938   
149                              House of Angostura          1830   
150                              Shirley Plantation          1638   
151            Bank of the Province of Buenos Aires          1822   
152                       Banco Nacional de Bolivia          1871   
153                         Casa da Moeda do Brasil          1694   
154                                           Famae          1811   
155                      Casa de Moneda de Colombia          1621   
156                                       Banks DIH          1840   
157                         Casa Nacional de Moneda          1565   
158                                  Cafe Brasilero          1877   
159                                  Haci

In [184]:
# Merge and filter to find countries with missing business data
new_all_countries = all_businesses.merge(countries, on="country_code", how="outer",  indicator=True)
new_all_countries

business  year_founded  \
0                                        Fiji Times        1869.0   
1                         J. Armando Bermúdez & Co.        1852.0   
2                                    Hamoud Boualem        1878.0   
3                    Communauté Électrique du Bénin        1968.0   
4                          Botswana Meat Commission        1965.0   
5                                       Air Burkina        1967.0   
6                                           Brarudi        1955.0   
7                  Cameroon Development Corporation        1947.0   
8                            Correios de Cabo Verde        1849.0   
9        Banque Internationale pour la Centrafrique        1946.0   
10                                       Cotontchad        1971.0   
11                      Central Bank of the Comoros        1981.0   
12    Société nationale des Chemins de fer du Congo        1889.0   
13   Development Bank of the Central African States        1975.0   
14                                         La Poste        1945.0   
15                          Ethio-Djibouti Railways        1901.0   
16                       Egyptian National Railways        1854.0   
17                       Guinea Ecuatorial Airlines        1996.0   
18                                   Asmara Brewery        1939.0   
19                        National Bank of Ethiopia        1906.0   
20                                   BGFIBank Group        1971.0   
21                                     Halco Mining        1962.0   
22                         Correios da Guiné-Bissau        1973.0   
23                                KCB Group Limited        1896.0   
24                          Central Bank of Lesotho        1978.0   
25                          National Port Authority        1921.0   
26                                        Umma Bank        1907.0   
27                                   Air Madagascar        1962.0   
28                  Malawi Broadcasting Corporation        1964.0   
29      Office de Radiodiffusion-Télévision du Mali        1957.0   
..                                              ...           ...   
165                                             NaN           NaN   
166                                             NaN           NaN   
167                                             NaN           NaN   
168                                             NaN           NaN   
169                                             NaN           NaN   
170                                             NaN           NaN   
171                                             NaN           NaN   
172                                             NaN           NaN   
173                                             NaN           NaN   
174                                             NaN           NaN   
175                                             NaN           NaN   
176                                             NaN           NaN   
177                                             NaN           NaN   
178                                             NaN           NaN   
179                                             NaN           NaN   
180                                             NaN           NaN   
181                                             NaN           NaN   
182                                             NaN           NaN   
183                                             NaN           NaN   
184                                             NaN           NaN   
185                                             NaN           NaN   
186                                             NaN           NaN   
187                                             NaN           NaN   
188                                             NaN           NaN   
189                                             NaN           NaN   
190                                             NaN           NaN   
191                                      

In [186]:
new_missing_countries = new_all_countries[new_all_countries["_merge"] != "both"]
new_missing_countries

business  year_founded category_code country_code  \
165      NaN           NaN           NaN          AGO   
166      NaN           NaN           NaN          ATG   
167      NaN           NaN           NaN          BHS   
168      NaN           NaN           NaN          ECU   
169      NaN           NaN           NaN          FSM   
170      NaN           NaN           NaN          GHA   
171      NaN           NaN           NaN          GMB   
172      NaN           NaN           NaN          GRD   
173      NaN           NaN           NaN          IRN   
174      NaN           NaN           NaN          KGZ   
175      NaN           NaN           NaN          KIR   
176      NaN           NaN           NaN          KNA   
177      NaN           NaN           NaN          MCO   
178      NaN           NaN           NaN          MDA   
179      NaN           NaN           NaN          MDV   
180      NaN           NaN           NaN          MHL   
181      NaN           NaN           NaN          NRU   
182      NaN           NaN           NaN          PLW   
183      NaN           NaN           NaN          PNG   
184      NaN           NaN           NaN          PRY   
185      NaN           NaN           NaN          PSE   
186      NaN           NaN           NaN          SLB   
187      NaN           NaN           NaN          SUR   
188      NaN           NaN           NaN          TJK   
189      NaN           NaN           NaN          TKM   
190      NaN           NaN           NaN          TLS   
191      NaN           NaN           NaN          TON   
192      NaN           NaN           NaN          TUV   
193      NaN           NaN           NaN          VCT   
194      NaN           NaN           NaN          WSM   

                              country      continent      _merge  
165                            Angola         Africa  right_only  
166               Antigua and Barbuda  North America  right_only  
167                           Bahamas  North America  right_only  
168                           Ecuador  South America  right_only  
169   Micronesia, Federated States of        Oceania  right_only  
170                             Ghana         Africa  right_only  
171                            Gambia         Africa  right_only  
172                           Grenada  North America  right_only  
173         Iran, Islamic Republic of           Asia  right_only  
174                        Kyrgyzstan           Asia  right_only  
175                          Kiribati        Oceania  right_only  
176             Saint Kitts and Nevis  North America  right_only  
177                            Monaco         Europe  right_only  
178              Moldova, Republic of         Europe  right_only  
179                          Maldives           Asia  right_only  
180                  Marshall Islands        Oceania  right_only  
181                             Nauru        Oceania  right_only  
182                             Palau        Oceania  right_only  
183                  Papua New Guinea        Oceania  right_only  
184                          Paraguay  South America  right_only  
185               Palestine, State of           Asia  right_only  
186                   Solomon Islands        Oceania  right_only  
187                          Suriname  South America  right_only  
188                        Tajikistan           Asia  right_only  
189                      Turkmenistan           Asia  right_only  
190                       Timor-Leste           Asia  right_only  
191                             Tonga        Oceania  right_only  
192                            Tuvalu        Oceania  right_only  
193  Saint Vincent and the Grenadines  North America  right_only  
194                             Samoa        Oceania  right_only

In [187]:
# Group by continent and create a "count_missing" column
count_missing = new_missing_countries.groupby("continent").agg({"country":"count"})
count_missing

country
continent             
Africa               3
Asia                 7
Europe               2
North America        5
Oceania             10
South America        3

In [188]:
count_missing.columns = ["count_missing"]
count_missing

count_missing
continent                   
Africa                     3
Asia                       7
Europe                     2
North America              5
Oceania                   10
South America              3

## 6. The oldest industries
<p>Remember our oldest business in the world, Kongō Gumi? </p>
<table>
<thead>
<tr>
<th style="text-align:right;"></th>
<th style="text-align:left;">business</th>
<th style="text-align:right;">year_founded</th>
<th style="text-align:left;">category_code</th>
<th style="text-align:left;">country_code</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">64</td>
<td style="text-align:left;">Kongō Gumi</td>
<td style="text-align:right;">578</td>
<td style="text-align:left;">CAT6</td>
<td style="text-align:left;">JPN</td>
</tr>
</tbody>
</table>
<p>We know Kongō Gumi was founded in the year 578 in Japan, but it's a little hard to decipher which industry it's in. Information about what the <code>category_code</code> column refers to is in <code>"datasets/categories.csv"</code>: </p>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">column</th>
<th>type</th>
<th>meaning</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>varchar</td>
<td>Code for the category of the business.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>varchar</td>
<td>Description of the business category.</td>
</tr>
</tbody>
</table>
<p>Let's use <code>categories.csv</code> to understand how many oldest businesses are in each category of industry.</p>

In [189]:
# Import categories.csv and merge to businesses
categories=pd.read_csv("datasets/categories.csv")
categories

category_code                           category
0           CAT1                        Agriculture
1           CAT2               Aviation & Transport
2           CAT3                  Banking & Finance
3           CAT4          Cafés, Restaurants & Bars
4           CAT5                       Conglomerate
5           CAT6                       Construction
6           CAT7                     Consumer Goods
7           CAT8                            Defense
8           CAT9  Distillers, Vintners, & Breweries
9          CAT10                             Energy
10         CAT11                   Food & Beverages
11         CAT12         Manufacturing & Production
12         CAT13                              Media
13         CAT14                            Medical
14         CAT15                             Mining
15         CAT16                     Postal Service
16         CAT17                             Retail
17         CAT18                 Telecommunications
18         CAT19                   Tourism & Hotels

In [190]:

businesses_categories=businesses.merge(categories,on='category_code',)
businesses_categories

business  year_founded  \
0                                   Hamoud Boualem          1878   
1                           M. M. Ispahani Limited          1820   
2                             George Steuart Group          1835   
3                    Corporación Multi Inversiones          1920   
4                                        Banks DIH          1840   
5                                   Hacienda Chuao          1660   
6                   Communauté Électrique du Bénin          1968   
7                                North Oil Company          1928   
8                              Electricite du Laos          1959   
9                       Petroleum Development Oman          1937   
10                        Botswana Meat Commission          1965   
11                Cameroon Development Corporation          1947   
12                                      Cotontchad          1971   
13                          Spinzar Cotton Company          1930   
14                               Casa de Ganaderos          1218   
15                              Shirley Plantation          1638   
16                                     Air Burkina          1967   
17   Société nationale des Chemins de fer du Congo          1889   
18                         Ethio-Djibouti Railways          1901   
19                      Egyptian National Railways          1854   
20                      Guinea Ecuatorial Airlines          1996   
21                         National Port Authority          1921   
22                                  Air Madagascar          1962   
23                      Beira Railroad Corporation          1892   
24                             Dakar–Niger Railway          1924   
25                                  Air Seychelles          1977   
26                                      Swazi Rail          1963   
27             Azerbaijan Caspian Shipping Company          1858   
28                                     Tashi Group          1959   
29                       Myanmar National Airlines          1948   
..                                             ...           ...   
133                        Casa da Moeda do Brasil          1694   
134                     Casa de Moneda de Colombia          1621   
135                                           BMMI          1883   
136                        Hua Ho Department Store          1947   
137                               M.H. Alshaya Co.          1890   
138                           Hudson's Bay Company          1670   
139             Ma Yu Ching's Bucket Chicken House          1153   
140                                 Café Abu Salem          1914   
141                                        Bakdash          1885   
142                    St. Peter Stifts Kulinarium           803   
143                                     Sean's Bar           900   
144                                 Cafe Brasilero          1877   
145                                         Pindad          1808   
146                                     Arsenal AD          1878   
147                                          Famae          1811   
148                                     Kongō Gumi           578   
149                               House of Alireza          1845   
150                                 House of Habib          1841   
151         Salam International Investment Limited          1952   
152                                  B.Grimm Group          1878   
153                     Petrodvorets Watch Factory          1721   
154                                        Fiskars          1649   
155                       Tutunski kombinat Prilep          1873   
156                                 KT Corporation          1885   
157                                     ALBtelecom          1912   
158                                      Raeapteek          1422   
159                                Gostilna Gastuž          1467   
160                           

In [191]:
# Create a DataFrame which lists the number of oldest businesses in each category
count_business_cats=businesses_categories.groupby('category').agg({'business':'count'})
count_business_cats

business
category                                   
Agriculture                               6
Aviation & Transport                     19
Banking & Finance                        37
Cafés, Restaurants & Bars                 6
Conglomerate                              3
Construction                              2
Consumer Goods                            3
Defense                                   3
Distillers, Vintners, & Breweries        22
Energy                                    4
Food & Beverages                          6
Manufacturing & Production               15
Media                                     7
Medical                                   1
Mining                                    3
Postal Service                           16
Retail                                    4
Telecommunications                        2
Tourism & Hotels                          4

In [192]:
# Create a DataFrame which lists the cumulative years that businesses from each category have been operating
years_business_cats=businesses_categories.groupby('category').agg({'year_founded':'sum'})
years_business_cats

year_founded
category                                       
Agriculture                               10669
Aviation & Transport                      36598
Banking & Finance                         70302
Cafés, Restaurants & Bars                  8532
Conglomerate                               5671
Construction                               2423
Consumer Goods                             5243
Defense                                    5497
Distillers, Vintners, & Breweries         37292
Energy                                     7792
Food & Beverages                          10953
Manufacturing & Production                21854
Media                                     13663
Medical                                    1422
Mining                                     5123
Postal Service                            29113
Retail                                     7390
Telecommunications                         3797
Tourism & Hotels                           6051

In [193]:
count_business_cats.columns=['count']
years_business_cats.columns= ['total_years_in_business']
display(count_business_cats.head(), years_business_cats.head())

count
category                        
Agriculture                    6
Aviation & Transport          19
Banking & Finance             37
Cafés, Restaurants & Bars      6
Conglomerate                   3

total_years_in_business
category                                          
Agriculture                                  10669
Aviation & Transport                         36598
Banking & Finance                            70302
Cafés, Restaurants & Bars                     8532
Conglomerate                                  5671

## 7. Restaurant representation
<p>No matter how we measure it, looks like Banking and Finance is an excellent industry to be in if longevity is our goal! Let's zoom in on another industry: cafés, restaurants, and bars. Which restaurants in our dataset have been around since before the year 1800?</p>

In [195]:
businesses_categories

business  year_founded  \
0                                   Hamoud Boualem          1878   
1                           M. M. Ispahani Limited          1820   
2                             George Steuart Group          1835   
3                    Corporación Multi Inversiones          1920   
4                                        Banks DIH          1840   
5                                   Hacienda Chuao          1660   
6                   Communauté Électrique du Bénin          1968   
7                                North Oil Company          1928   
8                              Electricite du Laos          1959   
9                       Petroleum Development Oman          1937   
10                        Botswana Meat Commission          1965   
11                Cameroon Development Corporation          1947   
12                                      Cotontchad          1971   
13                          Spinzar Cotton Company          1930   
14                               Casa de Ganaderos          1218   
15                              Shirley Plantation          1638   
16                                     Air Burkina          1967   
17   Société nationale des Chemins de fer du Congo          1889   
18                         Ethio-Djibouti Railways          1901   
19                      Egyptian National Railways          1854   
20                      Guinea Ecuatorial Airlines          1996   
21                         National Port Authority          1921   
22                                  Air Madagascar          1962   
23                      Beira Railroad Corporation          1892   
24                             Dakar–Niger Railway          1924   
25                                  Air Seychelles          1977   
26                                      Swazi Rail          1963   
27             Azerbaijan Caspian Shipping Company          1858   
28                                     Tashi Group          1959   
29                       Myanmar National Airlines          1948   
..                                             ...           ...   
133                        Casa da Moeda do Brasil          1694   
134                     Casa de Moneda de Colombia          1621   
135                                           BMMI          1883   
136                        Hua Ho Department Store          1947   
137                               M.H. Alshaya Co.          1890   
138                           Hudson's Bay Company          1670   
139             Ma Yu Ching's Bucket Chicken House          1153   
140                                 Café Abu Salem          1914   
141                                        Bakdash          1885   
142                    St. Peter Stifts Kulinarium           803   
143                                     Sean's Bar           900   
144                                 Cafe Brasilero          1877   
145                                         Pindad          1808   
146                                     Arsenal AD          1878   
147                                          Famae          1811   
148                                     Kongō Gumi           578   
149                               House of Alireza          1845   
150                                 House of Habib          1841   
151         Salam International Investment Limited          1952   
152                                  B.Grimm Group          1878   
153                     Petrodvorets Watch Factory          1721   
154                                        Fiskars          1649   
155                       Tutunski kombinat Prilep          1873   
156                                 KT Corporation          1885   
157                                     ALBtelecom          1912   
158                                      Raeapteek          1422   
159                                Gostilna Gastuž          1467   
160                           

In [196]:
# Filter using .query() for CAT4 businesses founded before 1800; sort results
old_restaurants = businesses_categories.query('year_founded < 1800 and category_code == "CAT4"')
old_restaurants

business  year_founded category_code  \
139  Ma Yu Ching's Bucket Chicken House          1153          CAT4   
142         St. Peter Stifts Kulinarium           803          CAT4   
143                          Sean's Bar           900          CAT4   

    country_code                   category  
139          CHN  Cafés, Restaurants & Bars  
142          AUT  Cafés, Restaurants & Bars  
143          IRL  Cafés, Restaurants & Bars

In [197]:
# Sort the DataFrame
old_restaurants = old_restaurants.sort_values('year_founded')
old_restaurants

business  year_founded category_code  \
142         St. Peter Stifts Kulinarium           803          CAT4   
143                          Sean's Bar           900          CAT4   
139  Ma Yu Ching's Bucket Chicken House          1153          CAT4   

    country_code                   category  
142          AUT  Cafés, Restaurants & Bars  
143          IRL  Cafés, Restaurants & Bars  
139          CHN  Cafés, Restaurants & Bars

## 8. Categories and continents
<p>St. Peter Stifts Kulinarium is old enough that the restaurant is believed to have served Mozart - and it would have been over 900 years old even when he was a patron! Let's finish by looking at the oldest business in each category of commerce for each continent. </p>

In [199]:
# Merge all businesses, countries, and categories together
businesses_categories_countries = businesses_categories.merge(countries,on='country_code')
businesses_categories_countries

business  year_founded  \
0                                   Hamoud Boualem          1878   
1                           M. M. Ispahani Limited          1820   
2                             George Steuart Group          1835   
3                    Corporación Multi Inversiones          1920   
4                                        Banks DIH          1840   
5                                   Hacienda Chuao          1660   
6                   Communauté Électrique du Bénin          1968   
7                                North Oil Company          1928   
8                              Electricite du Laos          1959   
9                       Petroleum Development Oman          1937   
10                        Botswana Meat Commission          1965   
11                Cameroon Development Corporation          1947   
12                                      Cotontchad          1971   
13                          Spinzar Cotton Company          1930   
14                               Casa de Ganaderos          1218   
15                              Shirley Plantation          1638   
16                                     Air Burkina          1967   
17   Société nationale des Chemins de fer du Congo          1889   
18                         Ethio-Djibouti Railways          1901   
19                      Egyptian National Railways          1854   
20                      Guinea Ecuatorial Airlines          1996   
21                         National Port Authority          1921   
22                                  Air Madagascar          1962   
23                      Beira Railroad Corporation          1892   
24                             Dakar–Niger Railway          1924   
25                                  Air Seychelles          1977   
26                                      Swazi Rail          1963   
27             Azerbaijan Caspian Shipping Company          1858   
28                                     Tashi Group          1959   
29                       Myanmar National Airlines          1948   
..                                             ...           ...   
133                        Casa da Moeda do Brasil          1694   
134                     Casa de Moneda de Colombia          1621   
135                                           BMMI          1883   
136                        Hua Ho Department Store          1947   
137                               M.H. Alshaya Co.          1890   
138                           Hudson's Bay Company          1670   
139             Ma Yu Ching's Bucket Chicken House          1153   
140                                 Café Abu Salem          1914   
141                                        Bakdash          1885   
142                    St. Peter Stifts Kulinarium           803   
143                                     Sean's Bar           900   
144                                 Cafe Brasilero          1877   
145                                         Pindad          1808   
146                                     Arsenal AD          1878   
147                                          Famae          1811   
148                                     Kongō Gumi           578   
149                               House of Alireza          1845   
150                                 House of Habib          1841   
151         Salam International Investment Limited          1952   
152                                  B.Grimm Group          1878   
153                     Petrodvorets Watch Factory          1721   
154                                        Fiskars          1649   
155                       Tutunski kombinat Prilep          1873   
156                                 KT Corporation          1885   
157                                     ALBtelecom          1912   
158                                      Raeapteek          1422   
159                                Gostilna Gastuž          1467   
160                           

In [200]:
# Sort businesses_categories_countries from oldest to most recent
businesses_categories_countries = businesses_categories_countries.sort_values('year_founded')
businesses_categories_countries

business  year_founded  \
148                                      Kongō Gumi           578   
142                     St. Peter Stifts Kulinarium           803   
44                            Staffelter Hof Winery           862   
127                               Monnaie de Paris            864   
126                                  The Royal Mint           886   
143                                      Sean's Bar           900   
128                          Marinelli Bell Foundry          1040   
41                                 Affligem Brewery          1074   
125                                     Munke Mølle          1135   
139              Ma Yu Ching's Bucket Chicken House          1153   
14                                Casa de Ganaderos          1218   
160                                 Gasthof Sternen          1230   
112                               Bochnia Salt Mine          1248   
131                            Drohobych salt plant          1250   
129                                   Kremnica Mint          1328   
49                                The Brand Brewery          1340   
130                                 Skyllbergs bruk          1346   
43                                  Pivovar Broumov          1348   
158                                       Raeapteek          1422   
159                                 Gostilna Gastuž          1467   
48                                           Mousel          1511   
71                      CTT-Correios de Portugal SA          1520   
132                     La Casa de Moneda de México          1534   
107                         Casa Nacional de Moneda          1565   
161                              Çemberlitaş Hamamı          1584   
46                                        Cēsu Alus          1590   
98                                     Piraeus Bank          1606   
134                      Casa de Moneda de Colombia          1621   
15                               Shirley Plantation          1638   
70                                     Posten Norge          1647   
..                                              ...           ...   
136                         Hua Ho Department Store          1947   
29                        Myanmar National Airlines          1948   
151          Salam International Investment Limited          1952   
90                        National Bank of Cambodia          1954   
35                                          Brarudi          1955   
114     Office de Radiodiffusion-Télévision du Mali          1957   
8                               Electricite du Laos          1959   
28                                      Tashi Group          1959   
115         Office of Radio and Television of Niger          1960   
32                                  Yemenia Airways          1962   
110                                    Halco Mining          1962   
22                                   Air Madagascar          1962   
26                                       Swazi Rail          1963   
113                 Malawi Broadcasting Corporation          1964   
10                         Botswana Meat Commission          1965   
16                                      Air Burkina          1967   
6                    Communauté Électrique du Bénin          1968   
77                                   BGFIBank Group          1971   
12                                       Cotontchad          1971   
59                         Correios da Guiné-Bissau          1973   
81                       Central Bank of Mauritania          1973   
84            Central Bank of São Tomé and Príncipe          1975   
75   Development Bank of the Central African States          1975   
25                                   Air Seychelles          1977   
79                          Central Bank of Lesotho          1978   
74                      Central Bank of the Comoros          1981   
109                          European Tru

In [201]:
# Create the oldest by continent and category DataFrame
oldest_by_continent_category =  businesses_categories_countries.groupby(["continent", "category"]).agg({"year_founded":"min"})
oldest_by_continent_category

year_founded
continent     category                                       
Africa        Agriculture                                1947
              Aviation & Transport                       1854
              Banking & Finance                          1892
              Distillers, Vintners, & Breweries          1933
              Energy                                     1968
              Food & Beverages                           1878
              Manufacturing & Production                 1820
              Media                                      1943
              Mining                                     1962
              Postal Service                             1772
Asia          Agriculture                                1930
              Aviation & Transport                       1858
              Banking & Finance                          1830
              Cafés, Restaurants & Bars                  1153
              Conglomerate                               1841
              Construction                                578
              Defense                                    1808
              Distillers, Vintners, & Breweries          1853
              Energy                                     1928
              Food & Beverages                           1820
              Manufacturing & Production                 1736
              Media                                      1931
              Mining                                     1913
              Postal Service                             1800
              Retail                                     1883
              Telecommunications                         1885
              Tourism & Hotels                           1584
Europe        Agriculture                                1218
              Banking & Finance                          1606
              Cafés, Restaurants & Bars                   803
              Consumer Goods                             1649
              Defense                                    1878
              Distillers, Vintners, & Breweries           862
              Manufacturing & Production                  864
              Media                                      1999
              Medical                                    1422
              Mining                                     1248
              Postal Service                             1520
              Telecommunications                         1912
              Tourism & Hotels                           1230
North America Agriculture                                1638
              Aviation & Transport                       1870
              Banking & Finance                          1891
              Distillers, Vintners, & Breweries          1703
              Food & Beverages                           1920
              Manufacturing & Production                 1534
              Media                                      1909
              Retail                                     1670
              Tourism & Hotels                           1770
Oceania       Banking & Finance                          1861
              Postal Service                             1809
South America Banking & Finance                          1565
              Cafés, Restaurants & Bars                  1877
              Defense                                    1811
              Food & Beverages                           1660
              Manufacturing & Production                 1621